server/main_server/src/rodsServer.cpp

```diff
@@ -462,11 +469,11 @@ namespace
         signal(SIGPIPE, SIG_IGN);
 #ifdef osx_platform
         signal(SIGINT, (sig_t) serverExit);
-        signal(SIGHUP, (sig_t) serverExit);
+        signal(SIGHUP, (sig_t) reload_next_loop);
```

Comment:
Consider changing `reload_next_loop`.

For example: `set_reload_server_config_flag`